src/Chapter3.hs

```diff
@@ -910,6 +1073,20 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append x y = x ++ y
+
+newtype Gold = Gold Int
+
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold x) (Gold y) = Gold (x+y)
+
+-- didn't quite understand what is required in this task :/
+--instance Append (Maybe a) where
+--    append :: (Maybe a) -> (Maybe a) -> (Maybe a)
```

Comment:
Yes, here is exactly the function that needs to be written.
Below you absolutely correctly cover the case when both of the arguments are `Just`s. As maybe id a sum type, there are 4 combinations that could potentially be:

```haskell
Just x  <->  Just y
Nothing <-> Just y
Just x  <-> Nothing
Nothing <-> Nothing
```
So the function pattern matching should cover them all.
Let me know if it makes more sense now, or you have a question about the meaning of such instance 🙂

Reply:
Hi @vrom911! Thanks for the feedback 😊 Yes, it does make sense now, I have fixed the function to work with all the cases.